app/src/main/scala/org/alephium/explorer/persistence/queries/TokenQueries.scala

```diff
@@ -110,6 +111,22 @@ object TokenQueries extends StrictLogging {
       #${pagination.query}
     """.asAS[TokenId]
 
+  def listAddressTokensBalanceAction(address: Address, pagination: Pagination)(
+      implicit ec: ExecutionContext): DBActionSR[TokenBalance] =
+    for {
+      tokens <- listAddressTokensAction(address, pagination)
+      balances <- DBIOAction.sequence(tokens.map { token =>
+        getTokenBalanceAction(address, token).map { balance =>
+          (token, balance)
+        }
+      })
+    } yield {
```

Comment:
Since the default limit here is `100`, this query could dispatch `100` queries for each request. Can we reduce this to fetch the data required in `1` or `2` queries?

If we don't wanna `join` then `getTokenBalanceAction` can query multiple tokens in a single query?

Reply:
I confess that seeing our [getTokenBalance query](https://github.com/alephium/explorer-backend/blob/07c05f2e1e420b369359792db1dfd59293444b73/app/src/main/scala/org/alephium/explorer/persistence/queries/TokenQueries.scala#L47-L65) made me go for the simplest solution :sweat_smile: 
Nevertheless I did try to think of something going your way and I'm not sure how as we have to join on the inputs anyway. 
I'll give a new try.